src/storage-controller/src/lib.rs

```diff
@@ -2982,19 +3018,21 @@ where
         &mut self,
         id: GlobalId,
         introspection_type: IntrospectionType,
+        recent_upper: Antichain<T>,
+        write_handle: Option<&mut WriteHandle<SourceData, (), T, Diff>>,
     ) -> Result<(), StorageError<T>> {
         tracing::info!(%id, ?introspection_type, "preparing introspection collection for writes");
 
         match introspection_type {
             IntrospectionType::ShardMapping => {
-                self.initialize_shard_mapping().await;
+                // Done by the `self.append_shard_mappings` call.
```

Comment:
Don't we still need this? The `append_shard_mappings` call in `create_collections` only does it for new collections, and `initialize_shard_mapping()` made sure we write down the collections that were already known at the time at which we "create" the shard-mappings shard.

Reply:
Yeah, not sure how we want this to work. I removed this because they were getting double registered. It seems that before, during initial boot we were hitting the following short-circuit when `append_shard_mappings` got called as part of `create_collections_for_bootstrap`. But this PR shifts the order slightly of things in `create_collections_for_bootstrap` and ends up putting `ShardMapping` in the `introspection_ids` map before `append_shard_mappings` gets called. This resulted in them getting added twice, once from `append_shard_mappings` and once from `initialize_shard_mapping`

```
        let id = match self
            .introspection_ids
            .lock()
            .expect("poisoned")
            .get(&IntrospectionType::ShardMapping)
        {
            Some(id) => *id,
            _ => return,
        };
```

Reply:
(We hashed this out in a huddle. Decision was to make this short-circuit a panic if that's possible, or add a comment if not.)